mmv1/products/gkeonprem/VmwareCluster.yaml

```diff
@@ -0,0 +1,673 @@
+                        description: Hostname of the machine. VM's name will be used
+                          if this field is empty.
+      - !ruby/object:Api::Type::NestedObject
+        name: "dhcpIpConfig"
+        description: Configuration settings for a DHCP IP configuration.
+        exactly_one_of:
+          - network_config.0.static_ip_config
+          - network_config.0.dhcp_ip_config
+        default_from_api: true
+        properties:
+          - !ruby/object:Api::Type::Boolean
+            name: enabled
+            description: |
+              enabled is a flag to mark if DHCP IP allocation is 
+              used for VMware user clusters.
+            required: true
+      - !ruby/object:Api::Type::String
+        name: "vcenterNetwork"
+        description: vcenter_network specifies vCenter network name. Inherited from
+          the admin cluster.
+        output: true
+      - !ruby/object:Api::Type::NestedObject
+        name: "hostConfig"
+        description: Represents common network settings irrespective of the host's
+          IP address.
+        default_from_api: true
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: "dnsServers"
+            description: DNS servers.
+            item_type: Api::Type::String
+          - !ruby/object:Api::Type::Array
+            name: "ntpServers"
+            description: NTP servers.
+            item_type: Api::Type::String
+          - !ruby/object:Api::Type::Array
+            name: "dnsSearchDomains"
+            description: DNS search domains.
+            item_type: Api::Type::String
+      - !ruby/object:Api::Type::NestedObject
+        name: "controlPlaneV2Config"
+        description: Configuration for control plane V2 mode.
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: "controlPlaneIpBlock"
+            description: Static IP addresses for the control plane nodes.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: "netmask"
+                description: The netmask used by the VMware User Cluster.
+              - !ruby/object:Api::Type::String
+                name: "gateway"
+                description: The network gateway used by the VMware User Cluster.
+              - !ruby/object:Api::Type::Array
+                name: "ips"
+                description: The node's network configurations used by the VMware
+                  User Cluster.
+                item_type: !ruby/object:Api::Type::NestedObject
+                  properties:
+                    - !ruby/object:Api::Type::String
+                      name: "ip"
+                      description: IP could be an IP address (like 1.2.3.4) or a CIDR
+                        (like 1.2.3.0/24).
+                    - !ruby/object:Api::Type::String
+                      name: "hostname"
+                      description: Hostname of the machine. VM's name will be used
```

Comment:
When this is sent as empty will the API return the VM name, or return empty? We'll need default_from_api if the VM name gets returned.

Reply:
Yes, default_from_api is added for the server-side defaulting logic. Thanks for checking!